datafusion/optimizer/src/expr_simplifier.rs

```diff
@@ -15,17 +15,24 @@
 // specific language governing permissions and limitations
 // under the License.
 
-//! Expression simplifier
+//! Expression simplification API
```

Comment:
This PR looks bigger than it really is -- it moves a bunch of code around and there are a significant number of docstrings